spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -465,6 +465,15 @@ public void setObservationRegistry(ObservationRegistry observationRegistry) {
 		this.observationRegistry = observationRegistry;
 	}
 
+	/**
+	 * Return the {@link ObservationRegistry} used by the template.
+	 * @return the observation registry
+	 * @since 3.3.5
```

Comment:
We can potentially backport this to 3.2.x. If that's the case, `@since` tag needs to be updates.

Reply:
Fair point, I can adapt.

Reply:
Updated to 3.2.9